src/main/java/seedu/address/logic/util/LessonModificationUtil.java

```diff
@@ -0,0 +1,110 @@
+import seedu.address.model.lesson.NumberOfOccurrences;
+import seedu.address.model.lesson.Venue;
+import seedu.address.model.student.Student;
+
+/**
+ * Contains utility methods for modifying {@code Attendances} in {@code Lessons}.
+ */
+public class LessonModificationUtil {
+
+    /**
+     * Adds all attendances in the target lesson to the new lesson.
+     * The {@code attendanceToAdd} is added to the lesson.
+     *
+     * @throws CommandException if the {@code attendanceToAdd} already exists.
+     */
+    public static Lesson createModifiedLessonWithAddedAttendance(
+            Lesson targetLesson, Student targetStudent, Week targetWeek, Attendance attendanceToAdd)
+            throws CommandException {
+        assert targetLesson != null;
+        assert targetStudent != null;
+        assert targetWeek != null;
+        assert attendanceToAdd != null;
+
+        if (targetLesson.getAttendanceRecordList().hasAttendance(targetStudent, targetWeek)) {
+            throw new CommandException(MESSAGE_DUPLICATE_ATTENDANCE);
+        }
+
+        List<AttendanceRecord> attendanceRecords = targetLesson.getAttendanceRecordList().getAttendanceRecordList();
+        Map<UUID, Attendance> record = targetLesson.getAttendanceRecordList()
+                .getAttendanceRecord(targetWeek).getAttendanceRecord();
+        Map<UUID, Attendance> updatedRecord = new HashMap<>(record);
+        updatedRecord.put(targetStudent.getUuid(), attendanceToAdd);
+
+        assert updatedRecord.size() == record.size() + 1;
+
+        AttendanceRecord updatedAttendanceRecord = new AttendanceRecord(updatedRecord);
+        List<AttendanceRecord> updatedAttendanceRecords = new ArrayList<>(attendanceRecords);
+        updatedAttendanceRecords.set(targetWeek.getZeroBasedWeekIndex(), updatedAttendanceRecord);
+        AttendanceRecordList updatedAttendanceRecordList = new AttendanceRecordList(updatedAttendanceRecords);
+
+        // unchanged lesson fields
+        LocalTime startTime = targetLesson.getStartTime();
+        LocalTime endTime = targetLesson.getEndTime();
+        Day day = targetLesson.getDay();
+        NumberOfOccurrences numberOfOccurrences = targetLesson.getNumberOfOccurrences();
+        Venue venue = targetLesson.getVenue();
+
+        return new Lesson(startTime, endTime, day, numberOfOccurrences, venue, updatedAttendanceRecordList);
+    }
+
+    /**
+     * Adds all attendances in the target lesson to the new lesson.
+     * The attendance of the {@code targetStudent} in the {@code targetWeek} is deleted from the lesson.
+     *
+     * @throws CommandException if the attendance of the {@code targetStudent} in the {@code targetWeek} does not exist.
+     */
+    public static Lesson createModifiedLessonWithDeletedAttendance(
+            Lesson targetLesson, Student targetStudent, Week targetWeek) throws CommandException {
+        assert targetLesson != null;
+        assert targetStudent != null;
+        assert targetWeek != null;
```

Comment:
Same as above. I notice that this occurs in the other refactored classes as well.